### PR TITLE
Adding or update initial value by class method in OspSystemStructure

### LINKED ISSUE #4: add a method to add an initial value to the system structure

### PR DESCRIPTION
**Pull request by kevinksyTRD**
closes #4

### CODE REVIEW

File: test/test_system_configuration.py [hunk 1]
```diff
@@ -30,6 +36,38 @@ def create_a_random_name(length: int):
...
+        target=OspVariableEndpoint(simulator=target_comp.name, name=create_a_random_name(5)),
+        group=group
+    )
+
+
+def create_a_random_pair_of_endpoints_for_sig_conn(
+        simulators: List[OspSimulator],
+        functions: OspFunctions,
+        group: bool
+) -> EndpointPair:
```
Comment (tibnor) [inline test/test_system_configuration.py:52]: A comment here could be useful. Describing that 50% of the time the endpoint is returned with source=OspVariable and target=OspSignal. In the other 50% this is reversed.

File: test/test_system_configuration.py [hunk 2]
```diff
@@ -498,3 +536,296 @@ def test_osp_system_structure():
...
     assertEqual(dict_xml, dict_xml_ref)
+
+
+def test_adding_updating_deleting_a_initial_values():
+    # create from a dict_xml
+    obj = OspSystemStructure(xml_source=PATH_TO_TEST_SYSTEM_STRUCTURE)
+
+    # Test failure by adding with a wrong component name
+    init_value = OspInitialValue(variable='new_variable', value=10.0)
+    with pytest.raises(TypeError):
```
Comment (tibnor) [inline test/test_system_configuration.py:552]: This could have been a separate test

File: test/test_system_configuration.py [hunk 3]
```diff
@@ -498,3 +536,296 @@ def test_osp_system_structure():
...
+        obj.add_update_initial_value(
+            component_name='This is not a correct name',
+            init_value=init_value
+        )
+
+    # Test adding a new variable
```
Comment (tibnor) [inline test/test_system_configuration.py:558]: Consider splitting each of the commented blocks into separate tests or subTests using test class

File: pyOSPParser/system_configuration.py [hunk 4]
```diff
@@ -124,6 +162,16 @@ class OspInitialValue(OspSystemStructureAbstract):
     _required_keys = ['variable', 'value']
 
     def __init__(self, dict_xml=None, **kwargs):
+        """
+        Constructor for OspInitialValue. One should provide 'variable (str)' and
```
Comment (tibnor) [inline pyOSPParser/system_configuration.py]: I don't understand this comment. Does dict_xml need to include the keys 'variable' and 'value' or should this be called with variable=..., value=...., dict_xml=.... If the later case, why is this not in the included in the signature? Consider also to add a validation with raising a ValueError.

File: pyOSPParser/system_configuration.py [hunk 5]
```diff
@@ -124,6 +162,16 @@ class OspInitialValue(OspSystemStructureAbstract):
...
+        'value (float, int, str, bool)'. Otherwise, one can provide dictionary that
+        contains the information.
+
+        Args:
+            variable
```
Comment (tibnor) [inline pyOSPParser/system_configuration.py]: These arguments are not in the signature, why?

**Review by tibnor — COMMENTED** [review]
I think you have done a great job commenting the module! It is much easier to understand the interface now! 🥇 

I think the PR is a too long for me to be able to understand it. You could for instance have done the reformating in another PR.

Consider also to add pylint to the automatic check.

File: test/test_system_configuration.py (hunk 3 above)
Comment (kevinksyTRD) [reply test/test_system_configuration.py:558]: Most of tests are linked with data or setup. I need to learn how to use fixture. Or I could use TestCase class in unittest package.

File: pyOSPParser/system_configuration.py (hunk 4 above)
Comment (kevinksyTRD) [reply pyOSPParser/system_configuration.py]: You need to provide either dict_xml or a set of arguments: 'variable' and 'value'. This should be an easier sentence to understand. Validation is in the parent class init.

File: pyOSPParser/system_configuration.py (hunk 5 above)
Comment (kevinksyTRD) [reply pyOSPParser/system_configuration.py]: These should be recognized in **kwargs and validated in the init using _required_keys attribute.

**Pull request merged**